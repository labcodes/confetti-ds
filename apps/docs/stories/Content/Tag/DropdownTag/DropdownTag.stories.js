import React from "react";

import { TAG_COLORS } from "../../../../confetti-ds/src/constants";
import { DropdownTag as Component  } from "../../../../confetti-ds/src";

export default {
  title: "Content/Tag/Dropdown Tag/Dropdown Tag",
  component: Component,
  argTypes: {
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
    icon: {
      control: false,
    },
  },
};

export const DropdownTag = (args) => <Component {...args} />;

DropdownTag.args = {
  text: "demo tag",
};
