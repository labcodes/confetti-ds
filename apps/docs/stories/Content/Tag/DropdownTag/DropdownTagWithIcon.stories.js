import React from "react";

import { ICON_TYPES, TAG_COLORS } from "../../../../confetti-ds/src/constants";
import { DropdownTag as Component  } from "../../../../confetti-ds/src";

export default {
  title: "Content/Tag/Dropdown Tag",
  component: Component,
  argTypes: {
    icon: {
      control: { type: "select", options: ICON_TYPES },
    },
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
  },
};

export const DropdownTagWithIcon = (args) => <Component {...args} />;

DropdownTagWithIcon.args = {
  text: "demo tag",
  icon: "coin",
};
