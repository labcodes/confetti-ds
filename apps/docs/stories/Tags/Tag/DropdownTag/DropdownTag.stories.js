import React from "react";

import { TAG_COLORS } from "../../../../../../packages/confetti-ds/src/constants";
import { DropdownTag as Component } from "../../../../../../packages/confetti-ds/src";

export default {
  title: "Tags/Tag/Dropdown Tag",
  component: Component,
  argTypes: {
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
    icon: {
      control: false,
    },
  },
};

export const DropdownTag = (args) => <Component {...args} />;

DropdownTag.args = {
  text: "demo tag",
};
