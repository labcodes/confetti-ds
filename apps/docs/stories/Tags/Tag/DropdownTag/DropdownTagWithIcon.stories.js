import React from "react";

import {
  ICON_TYPES,
  TAG_COLORS,
} from "../../../../../../packages/confetti-ds/src/constants";
import { DropdownTag as Component } from "../../../../../../packages/confetti-ds/src";

export default {
  title: "Tags/Tag/Dropdown Tag",
  component: Component,
  argTypes: {
    icon: {
      control: { type: "select", options: ICON_TYPES },
    },
    color: {
      control: { type: "select", options: TAG_COLORS },
    },
    skin: {
      control: { type: "inline-radio" },
    },
  },
};

export const DropdownTagWithIcon = (args) => <Component {...args} />;

DropdownTagWithIcon.args = {
  text: "demo tag",
  icon: "coin",
};
