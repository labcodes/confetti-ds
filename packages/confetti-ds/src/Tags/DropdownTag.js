import React from "react";
import PropTypes from "prop-types";

import AbstractTag from "./AbstractTag";
import Icon from "../Icon";
import { TAG_COLORS } from "../constants";

export default class DropdownTag extends React.Component {
  static propTypes = {
    /** This is the Tag's text. */
    text: PropTypes.string.isRequired,
    /** Type of the icon to be rendered. Won't render an icon if not passed to the component. */
    icon: PropTypes.string,
    /** Sets Tag's color. */
    color: PropTypes.oneOf(TAG_COLORS),
    /** Skin of the the rendered Tag. */
    skin: PropTypes.oneOf(["pale", "vivid"]),
    /** Sets an outline style. */
    isOutline: PropTypes.bool,
    /** Disables the Tag. Will be read by screen readers. When true, will override `disabled`. */
    ariaDisabled: PropTypes.bool,
    /** Disables the Tag. Won't be read by screen readers. */
    disabled: PropTypes.bool,
    /** Action to be executed when the Tag is clicked. */
    onClick: PropTypes.func,
  };

  static defaultProps = {
    icon: "",
    color: "",
    skin: "pale",
    isOutline: false,
    disabled: false,
    ariaDisabled: false,
    onClick: () => {},
  };

  icon = () => {
    const { icon } = this.props;
    return icon ? (
      <Icon
        type={icon}
        color="black-75"
        size="petit"
        className="lab-tag--left-icon"
      />
    ) : undefined;
  };

  dropdownIcon = () => (
    <span className="lab-tag__dropdown-icon-wrapper">
      <Icon
        type="dropdown-closed"
        color="black-75"
        size="petit"
        className="lab-tag--dropdown-icon"
      />
    </span>
  );

  render() {
    const {
      text,
      icon,
      color,
      skin,
      isOutline,
      disabled,
      ariaDisabled,
      onClick,
    } = this.props;
    return (
      <AbstractTag
        className={`lab-tag--dropdown${`${
          icon ? ` lab-tag--has-left-icon` : ""
        }`}`}
        text={text}
        icon={icon}
        color={color}
        skin={skin}
        isOutline={isOutline}
        disabled={!ariaDisabled && disabled}
        ariaDisabled={ariaDisabled}
        onClick={onClick}
        renderPrefix={this.icon()}
        renderSuffix={this.dropdownIcon()}
      />
    );
  }
}
