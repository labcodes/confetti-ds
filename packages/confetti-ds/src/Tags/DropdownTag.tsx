import React, { SyntheticEvent } from "react";

import AbstractTag from "./AbstractTag";
import Icon from "../Icon";
import { IconTypes, TagTypes } from "../constants";

export interface DropdownTagProps {
  /** This is the Tag's text. */
  text: string;
  /** Type of the icon to be rendered. Won't render an icon if not passed to the component. */
  icon?: IconTypes;
  /** Sets Tag's color. */
  color?: TagTypes;
  /** Skin of the rendered Tag. */
  skin?: "pale" | "vivid";
  /** Sets an outline style. */
  isOutline?: boolean;
  /** Disables the Tag. Will be read by screen readers. When true, will override `disabled`. */
  ariaDisabled?: boolean;
  /** Disables the Tag. Won't be read by screen readers. */
  disabled?: boolean;
  /** Action to be executed when the Tag is clicked. */
  onClick?: (event?: SyntheticEvent) => any;
}

export default function DropdownTag({
  text,
  icon,
  color,
  skin = "pale",
  isOutline = false,
  ariaDisabled = false,
  disabled = false,
  onClick = () => {},
}: DropdownTagProps) {
  const handleIcon = () =>
    icon ? <Icon type={icon} className="lab-tag--left-icon" /> : undefined;

  const dropdownIcon = () => (
    <span className="lab-tag__dropdown-icon-wrapper">
      <Icon type="dropdown-closed" className="lab-tag--dropdown-icon" />
    </span>
  );

  return (
    <AbstractTag
      className={`lab-tag--dropdown${`${
        icon ? ` lab-tag--has-left-icon` : ""
      }`}`}
      text={text}
      color={color}
      skin={skin}
      isOutline={isOutline}
      disabled={!ariaDisabled && disabled}
      ariaDisabled={ariaDisabled}
      onClick={onClick}
      renderPrefix={handleIcon()}
      renderSuffix={dropdownIcon()}
      tabIndex={0}
    />
  );
}
