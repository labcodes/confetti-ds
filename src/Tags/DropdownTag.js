import React from "react";
import PropTypes from "prop-types";

import AbstractTag from "./AbstractTag";
import Icon from "../Icon";
import { ICON_TYPES, TAG_COLORS } from "../constants";

/**
 *
 * @param text is the Tag's text.
 * @param icon is the type of the icon to be rendered. Won't render an icon if not passed to the component.
 * @param color sets Tag's color.
 * @param skin is the skin of the rendered Tag
 * @param isOutline sets an outline style.
 * @param disabled disables the Tag. Won't be read by screen readers.
 * @param ariaDisabled disables the Tag. Will be read by screen readers. When true, will override `disabled`.
 * @param onClick is the action to be executed when the Tag is clicked.
 * @returns {JSX.Element}
 * @constructor
 */
export default function DropdownTag({
  text,
  icon,
  color,
  skin,
  isOutline,
  disabled,
  ariaDisabled,
  onClick,
}) {
  // eslint-disable-next-line react/display-name
  const handleIcon = () =>
    icon ? (
      <Icon
        type={icon}
        color="black-75"
        size="petit"
        className="lab-tag--left-icon"
      />
    ) : undefined;

  const dropdownIcon = () => (
    <span className="lab-tag__dropdown-icon-wrapper">
      <Icon
        type="dropdown-closed"
        color="black-75"
        size="petit"
        className="lab-tag--dropdown-icon"
      />
    </span>
  );

  return (
    <AbstractTag
      className={`lab-tag--dropdown${`${
        icon ? ` lab-tag--has-left-icon` : ""
      }`}`}
      text={text}
      icon={icon}
      color={color}
      skin={skin}
      isOutline={isOutline}
      disabled={!ariaDisabled && disabled}
      ariaDisabled={ariaDisabled}
      onClick={onClick}
      renderPrefix={handleIcon()}
      renderSuffix={dropdownIcon()}
      tabIndex="0"
    />
  );
}

DropdownTag.propTypes = {
  /** This is the Tag's text. */
  text: PropTypes.string.isRequired,
  /** Type of the icon to be rendered. Won't render an icon if not passed to the component. */
  icon: PropTypes.oneOf(ICON_TYPES),
  /** Sets Tag's color. */
  color: PropTypes.oneOf(TAG_COLORS),
  /** Skin of the rendered Tag. */
  skin: PropTypes.oneOf(["pale", "vivid"]),
  /** Sets an outline style. */
  isOutline: PropTypes.bool,
  /** Disables the Tag. Will be read by screen readers. When true, will override `disabled`. */
  ariaDisabled: PropTypes.bool,
  /** Disables the Tag. Won't be read by screen readers. */
  disabled: PropTypes.bool,
  /** Action to be executed when the Tag is clicked. */
  onClick: PropTypes.func,
};

DropdownTag.defaultProps = {
  icon: undefined,
  color: undefined,
  skin: "pale",
  isOutline: false,
  disabled: false,
  ariaDisabled: false,
  onClick: () => {},
};
