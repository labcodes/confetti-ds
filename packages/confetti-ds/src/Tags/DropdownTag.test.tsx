import React from "react";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

import DropdownTag from "./DropdownTag";

describe("DropdownTag", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<DropdownTag text="Test render DropdownTag" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(<DropdownTag text="Test DropdownTag" />).html();
    expect(wrapper).toContain("lab-tag--dropdown");
    expect(wrapper).toContain("lab-icon--dropdown");
  });

  it("renders as disabled", async () => {
    const renderedComponent = renderer
      .create(<DropdownTag text="Test render disabled DropdownTag" disabled />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(
      <DropdownTag text="Test disabled DropdownTag" disabled />
    ).html();
    expect(wrapper).toContain("lab-tag--disabled");
  });

  it("render as outline", async () => {
    const renderedComponent = renderer
      .create(<DropdownTag text="Test render outline DropdownTag" isOutline />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(
      <DropdownTag text="Test outline DropdownTag" isOutline />
    ).html();
    expect(wrapper).toContain("lab-tag--outline");
  });

  it("renders with a pink color", async () => {
    const renderedComponent = renderer
      .create(<DropdownTag text="Test render black-75 DropdownTag" color="black-75" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();

    const wrapper = shallow(
      <DropdownTag text="Test pink DropdownTag" color="black-75" />
    ).html();
    expect(wrapper).toContain("lab-tag--black-75");
  });

  it("calls prop.onClick when clicked", async () => {
    const mockOnClick = jest.fn();
    const shallowDropdownTag = shallow(
      <DropdownTag text="Test render " onClick={mockOnClick} />
    );
    expect(mockOnClick.mock.calls.length).toEqual(0);
    shallowDropdownTag.simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });

  it("doesn't trigger onClick if ariaDisabled", async () => {
    const mockOnClick = jest.fn();
    const mountedComponent = mount(
      <DropdownTag ariaDisabled text="Test" onClick={mockOnClick} />
    );

    expect(mockOnClick).not.toBeCalled();
    mountedComponent.find("span").at(0).simulate("click");
    expect(mockOnClick).not.toBeCalled();
  });
});
