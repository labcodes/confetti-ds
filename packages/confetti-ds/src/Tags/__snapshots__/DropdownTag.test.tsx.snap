// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownTag render as outline 1`] = `
<span
  className="lab-tag lab-tag--dropdown lab-tag--outline lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  Test render outline DropdownTag
  <span
    className="lab-tag__dropdown-icon-wrapper"
  >
    <span
      className="lab-icon lab-icon--dropdown-closed lab-icon--black-75 lab-icon--petit lab-tag--dropdown-icon"
    />
  </span>
</span>
`;

exports[`DropdownTag renders as disabled 1`] = `
<span
  className="lab-tag lab-tag--dropdown lab-tag--disabled lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  Test render disabled DropdownTag
  <span
    className="lab-tag__dropdown-icon-wrapper"
  >
    <span
      className="lab-icon lab-icon--dropdown-closed lab-icon--black-75 lab-icon--petit lab-tag--dropdown-icon"
    />
  </span>
</span>
`;

exports[`DropdownTag renders with a pink color 1`] = `
<span
  className="lab-tag lab-tag--dropdown lab-tag--pink-pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  Test render pink DropdownTag
  <span
    className="lab-tag__dropdown-icon-wrapper"
  >
    <span
      className="lab-icon lab-icon--dropdown-closed lab-icon--black-75 lab-icon--petit lab-tag--dropdown-icon"
    />
  </span>
</span>
`;

exports[`DropdownTag renders with base props 1`] = `
<span
  className="lab-tag lab-tag--dropdown lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  Test render DropdownTag
  <span
    className="lab-tag__dropdown-icon-wrapper"
  >
    <span
      className="lab-icon lab-icon--dropdown-closed lab-icon--black-75 lab-icon--petit lab-tag--dropdown-icon"
    />
  </span>
</span>
`;
